fragalysis_api/xcimporter/align.py

```diff
@@ -99,34 +93,80 @@ def __get_length_and_resolution(self, file):
         for pp in ppb.build_peptides(structure):  # Retrieve length by looping through each chain in the protein
             seq_len += len(pp.get_sequence())
 
-        # using a functions from PDBParser parser class to get the resolution and protein id from the pdb file 
+        # using a functions from PDBParser parser class to get the resolution and protein id from the pdb file
         return pd.Series([structure.header['resolution'], seq_len, structure.id])
 
-    def _save_align(self, path_save):
+    def __get_pdb_file(self, pdb_name):
         """
-        Saves aligned structures as .pdb files
-        :param self:
-        :return aligned structures saved as .pdb files:
+        Returns the pdb file path for a given pdb name.
+        :param pdb_name: name of pdb
+        :return: file path to the pdb with the given name
         """
+
+        for file in self._get_files:
+            if pdb_name == os.path.splitext(os.path.basename(file))[0]:
+                return file
+
+    def __get_header(self, file):
+        """
+        Get headers (front and end) for a given pdb file.
+        :param file: pdb file
+        :return: front and end headers for given pdb file
+        """
+        with open(file) as handle:
+            switch = 0
+            header_front, header_end = [], []
+
+            for line in handle:
+
+                if line.startswith('ATOM'): switch = 1
+
+                if line.startswith('HETATM'): switch = 2
+
+                if switch == 0: header_front.append(line)
+
+                if (switch == 2) and not line.startswith('HETATM'): header_end.append(line)
+
+        return header_front, header_end
```

Comment:
Code climate has an issue with the 'Cognitive Complexity' of this function we should refactor it if possible